client/js/lounge.js

```diff
@@ -682,18 +657,47 @@ $(function() {
 			"quit",
 			"nick",
 			"mode",
+			"whois"
 		];
+		var type = msg.type;
 		if ($.inArray(type, ignore) !== -1){
 			return;
 		}
 
-		var badge = button.find(".badge");
-		if (badge.length !== 0) {
-			var i = (badge.data("count") || 0) + 1;
-			badge.data("count", i);
-			badge.html(i > 999 ? (i / 1000).toFixed(1) + "k" : i);
-			if (highlight || isQuery) {
-				badge.addClass("highlight");
+		var isQuery = button.hasClass("query");
+		var highlight = type.contains("highlight");
+		var message = type.contains("message");
+		if (highlight || isQuery || (options.notifyAllMessages && message)) {
+			if (options.notification) {
```

Comment:
So, if I'm reading this right, this will show a notification, badge and sound even if the tab is active?